lib/collection/src/collection/payload_index_schema.rs

```diff
@@ -84,4 +84,37 @@ impl Collection {
 
         Ok(result)
     }
+
+    /// Returns an arbitrary payload key used by `filter` which can be indexed but currently is not.
+    /// If this function returns `None` all indexable keys in `filter` are indexed.
+    pub fn has_filter_without_index(&self, filter: &Filter) -> Option<JsonPath> {
+        filter
+            .must
+            .iter()
+            .flatten()
+            .chain(filter.must_not.iter().flatten())
+            .chain(filter.should.iter().flatten())
+            .chain(filter.min_should.iter().flat_map(|i| &i.conditions))
+            .find_map(|condition| self.has_condition_without_index(condition))
+    }
+
+    fn has_condition_without_index(&self, condition: &Condition) -> Option<JsonPath> {
+        let has_key = |key: &JsonPath| self.payload_index_schema.read().schema.contains_key(key);
+
+        match condition {
+            Condition::Field(field) => {
+                if !has_key(&field.key) {
+                    Some(field.key.clone())
+                } else {
+                    None
+                }
+            }
+            Condition::Nested(nested) => self.has_filter_without_index(nested.filter()),
```

Comment:
It would be nice to see extra tests for nested fields, as they use relative keys